internal/healthchecks/error.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthchecks
+
+// Error classification
+const (
+	Api        = "API"
+	Connection = "CONNECTION"
+	Generic    = "GENERIC"
+	Port       = "PORT"
+	Permission = "PERMISSION"
+)
+
+type HealthCheckError struct {
+	Code         string
+	Class        string
+	Message      string
+	Action       string
+	ResourceLink string
+	IsFatal      bool
+	Err          error
```

Comment:
This can be removed, as it is only used in `Error()` and have `Error()` return `Message` instead.

It also allow for the usage of  `errors.Is()` in `healthchecks_test.go`